Reworked documentation of Soroban RPC methods using open-rpc-docs-react component to be consistent with Anchor API

MazurakIhor opened the pull request.

stellar-jenkins commented:
**Preview is available here:**<br>http://soroban-docs-pr603.previews.kube001.services.stellar-ops.com/

ElliotFriend commented:
Hi, @MazurakIhor Thanks SO much for putting this PR together! I've been really interested in getting our RPC spec into OpenRPC format recently, working to get the spec generated from the source code itself.

This PR is a really good step toward integrating the OpenRPC document(s) into our docusaurus site! We've had some other PRs merged in lately that have presented some conflicts on your branch. Could you resolve those for me, and push you changes back to your main branch?

stellar-jenkins commented:
**Preview is available here:**<br>http://soroban-docs-pr603.previews.kube001.services.stellar-ops.com/

MazurakIhor commented:
Hi, @ElliotFriend.
I merged latest main, resolved merge conflicts and updated descriptions in OpenRPC spec.

ElliotFriend commented on src/components/RpcMethod.tsx:
I don't know much about custom styles in docusaurus components, nor much about stoplight (which the `open-rpc-docs-react` package seems to use). But I'd be interested in getting adjusting some of the way the params/results are displayed.

![Screenshot 2023-10-18 at 11 07 44 AM](https://github.com/stellar/soroban-docs/assets/2024293/389816d3-9b14-4025-8886-6f8b1629bac8)

- Can we get rid of the numbers by the params?
- The param type feels strange so far away from the param name and underneath. Could it be moved? Just like it is with the results would make sense.
- The `font-size` in the results section seems a bit small. Could those be adjusted?

No worries if you're not the one who can tackle those, but just wanted to throw them out there.

ElliotFriend reviewed (commented):
Thanks for the updates! Here are some thoughts/questions that came to mind as I was looking through the changes

stellar-jenkins commented:
**Preview is available here:**<br>http://soroban-docs-pr603.previews.kube001.services.stellar-ops.com/

MazurakIhor commented:
@ElliotFriend 
I merged latest main and resolved merge conflicts.
Updated OpenRPC spec for getLedgerEntries RPC call.
Fixed TypeScript error.

Regarding styles: will try to adjust styles mentioned in your last comment, however in such case Soroban docs will differ from Stellar docs which were released recently.
Just an example:https://developers.stellar.org/api/anchor-platform/rpc/methods/do_stellar_payment/

ElliotFriend commented:
> Regarding styles: will try to adjust styles mentioned in your last comment, however in such case Soroban docs will differ from Stellar docs which were released recently.

I'm a dummy! I didn't know that the Stellar docs had incorporated this. No worries about the styles then. I think it's likely better to leave it as is.

Thanks!

ElliotFriend reviewed (approved):
Thanks again for the updates @MazurakIhor 

This is a great step toward automating the spec even more!
